schema_salad/tests/test_codegen_errors.py

```diff
@@ -0,0 +1,220 @@
+"""Tests of helpful error messages."""
+
+from pathlib import Path
+from typing import Any, MutableSequence, Optional, Union, cast
+from urllib.parse import urlparse
+
+import pytest
+
+import schema_salad.tests.cwl_v1_0 as cwl_v1_0
+from schema_salad.exceptions import ValidationException
+from schema_salad.utils import yaml_no_ts
+
+from .util import get_data
+
+
+def test_error_message1() -> None:
+    t = "test_schema/test1.cwl"
+    match = r"""^.*test1\.cwl:2:1:\s+Object\s+`.*test1\.cwl`\s+is\s+not\s+valid\s+because
+\s+tried\s+`Workflow`\s+but
+\s+\*\s+missing\s+required\s+field\s+`inputs`
+\s+\*\s+missing\s+required\s+field\s+`outputs`
+\s+\*\s+missing\s+required\s+field\s+`steps`"""
+    path = get_data("tests/" + t)
+    assert path
+    with pytest.raises(ValidationException, match=match):
+        load_document_by_uri(path)
+
+
+def test_error_message2() -> None:
+    t = "test_schema/test2.cwl"
+    match = r"""^.*test2\.cwl:2:1:\s+Field\s+`class`\s+contains\s+undefined\s+reference\s+to
+\s+`file://.+/schema_salad/tests/test_schema/xWorkflow`$"""
+    path = get_data("tests/" + t)
+    assert path
+    with pytest.raises(ValidationException, match=match):
+        load_document_by_uri(path)
+
+
```

Comment:
Do you think something like this is okay:
```
test_schema/test4.cwl:2:1: Object `test4.cwl` is not valid because
test_schema/test4.cwl:6:1:   the `outputs` field is not valid because:
test_schema/test4.cwl:7:3:     checking object `test4.cwl#bar`
                                 the `type` field is not valid because:
                                   Expected one of (list, str, dict) was int
```                                   
It's still a little different than the original validator but it got rid of `tried 'X' but Y` error messages (also replaced `<class 'int'>` with just `int`).

Reply:
Yes, that is much better.